pkg/autopilot/controller/setup.go

```diff
@@ -42,21 +42,23 @@ type SetupController interface {
 }
 
 type setupController struct {
-	log           *logrus.Entry
-	clientFactory apcli.FactoryInterface
-	k0sDataDir    string
-	enableWorker  bool
+	log              *logrus.Entry
+	clientFactory    apcli.FactoryInterface
+	k0sDataDir       string
+	enableWorker     bool
+	kubeletExtraArgs string
 }
 
 var _ SetupController = (*setupController)(nil)
 
 // NewSetupController creates a `SetupController`
-func NewSetupController(logger *logrus.Entry, cf apcli.FactoryInterface, k0sDataDir string, enableWorker bool) SetupController {
+func NewSetupController(logger *logrus.Entry, cf apcli.FactoryInterface, k0sDataDir, kubeletExtraArgs string, enableWorker bool) SetupController {
```

Comment:
We could store the kubelet's node name directly in the setup controller, instead of the whole extra args, and do the args parsing in the root controller instead.

Reply:
I decided to parse the args exactly where it was needed.